### PR TITLE
Makefile: Add BIND_GIT variable

### PR DESCRIPTION
**Pull request by vvoland**
Defining BIND_GIT will also bind mount the common .git directory into the dev container.
This makes it possible to run some hack/* scripts which rely on `git` when running the container in a git worktree.

**- How to verify it**
```terminal
$ git worktree add asdf
$ cd asdf
$ make BIND_GIT=1 validate-all
```

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -103,7 +103,14 @@ DOCKER_MOUNT := $(if $(DOCKER_MOUNT),$(DOCKER_MOUNT),-v /go/src/github.com/docke
 DOCKER_MOUNT_CACHE := -v docker-dev-cache:/root/.cache -v docker-mod-cache:/go/pkg/mod/
 DOCKER_MOUNT_CLI := $(if $(DOCKER_CLI_PATH),-v $(shell dirname $(DOCKER_CLI_PATH)):/usr/local/cli,)
 DOCKER_MOUNT_BASH_COMPLETION := $(if $(DOCKER_BASH_COMPLETION_PATH),-v $(shell dirname $(DOCKER_BASH_COMPLETION_PATH)):/usr/local/completion/bash,)
-DOCKER_MOUNT := $(DOCKER_MOUNT) $(DOCKER_MOUNT_CACHE) $(DOCKER_MOUNT_CLI) $(DOCKER_MOUNT_BASH_COMPLETION)
+
+ifdef BIND_GIT
+	# Gets the common .git directory (even from inside a git worktree)
+	GITDIR := $(shell realpath $(shell git rev-parse --git-common-dir))
+	MOUNT_GITDIR := $(if $(GITDIR),-v $(GITDIR):$(GITDIR))
```
Comment (thaJeztah) [inline Makefile:110]: Wondering if some quoting is needed in some of this; at least, considering if some of these could fail, and now produce some invalid paths 😅

Comment (vvoland) [reply Makefile]: Hmm right, added quotes so it doesn't fail for paths with spaces.

**Review by thaJeztah — APPROVED** [review]
generally LGTM

I must say that we should have a look at all options we have, as the list is growing rapidly, and I wonder if we can simplify things somehow.

LOL, I initially was confused as well ("isn't `BIND_DIR` already mounting all the things?" but realised that doesn't work with worktree)

**Review by austinvazquez — APPROVED** [review]

**Review by tianon — APPROVED** [review]
Agreed that this whole thing is pretty "byzantine" now but the use case here is pretty clear :+1:

**Comment by thaJeztah** [comment]
Thanks @tianon !

Adding "Byzantine" to my list of pretty words to describe ugly things 😂

**Pull request merged**